modules/apps/adaptive-media/adaptive-media-image-api/src/main/java/com/liferay/adaptive/media/image/optimizer/BaseAMImageOptimizer.java

```diff
@@ -0,0 +1,228 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.adaptive.media.image.optimizer;
+
+import com.liferay.adaptive.media.constants.AMOptimizeImagesBackgroundTaskConstants;
+import com.liferay.adaptive.media.image.configuration.AMImageConfigurationEntry;
+import com.liferay.adaptive.media.image.configuration.AMImageConfigurationHelper;
+import com.liferay.adaptive.media.image.mime.type.AMImageMimeTypeProvider;
+import com.liferay.adaptive.media.image.validator.AMImageValidator;
+import com.liferay.adaptive.media.processor.AMProcessor;
+import com.liferay.document.library.configuration.DLFileEntryConfigurationProvider;
+import com.liferay.document.library.constants.DLFileEntryConfigurationConstants;
+import com.liferay.document.library.kernel.model.DLFileEntry;
+import com.liferay.portal.kernel.backgroundtask.BackgroundTaskStatusMessageSender;
+import com.liferay.portal.kernel.backgroundtask.BackgroundTaskThreadLocal;
+import com.liferay.portal.kernel.backgroundtask.constants.BackgroundTaskConstants;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.messaging.Message;
+import com.liferay.portal.kernel.repository.model.FileEntry;
+import com.liferay.portal.kernel.repository.model.FileVersion;
+import com.liferay.portal.kernel.util.ArrayUtil;
+import com.liferay.portal.repository.liferayrepository.model.LiferayFileEntry;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Consumer;
+
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Marco Galluzzi
+ */
+public abstract class BaseAMImageOptimizer implements AMImageOptimizer {
+
+	@Override
+	public void optimize(long companyId) {
+		long previewableProcessorMaxSize =
```

Comment:
For consistency with other usages, why not name this variable `companyPreviewableProcessorMaxSize`?

Reply:
When you only need one type of value, from company, group or system, IMO there is no need to name the variable with it. This criterium has also been followed by Brian recently in a commit when he explicitly changed a variable to remove the scope.